ocaml/lambda/lambda.ml

```diff
@@ -1871,21 +1871,24 @@ let structured_constant_layout = function
   | Const_mixed_block _ | Const_block _ | Const_immstring _ -> Pvalue Pgenval
   | Const_float_array _ | Const_float_block _ -> Pvalue (Parrayval Pfloatarray)
 
+let rec layout_of_const_sort (c : Jkind.Sort.Const.t) : layout =
+  match c with
+  | Base Value -> layout_any_value
+  | Base Float64 -> layout_unboxed_float Pfloat64
+  | Base Float32 -> layout_unboxed_float Pfloat32
+  | Base Word -> layout_unboxed_nativeint
+  | Base Bits32 -> layout_unboxed_int32
+  | Base Bits64 -> layout_unboxed_int64
+  | Base Void -> assert false
```

Comment:
```suggestion
  | Base Void -> layout_unboxed_product []
```

?

Though, actually, I think I want

```
type base_layout =
  | Pvalue of value_kind
  | Punboxed_float of boxed_float
  | Punboxed_int of boxed_integer
  | Punboxed_vector of boxed_vector

type layout =
  | Ptop
  | Punboxed_product of base_layout iarray
  | Pbottom
```

and then to flatten the lists. This would be a big change, I think, and not (at all) for this PR. But could this be a good idea? Maybe we stick `Pvalue` also in `layout` because it's a common case and indirection is annoying... but the singleton list of values would be identical to value.

Reply:
Re void: Possibly soon! But I think not in this PR - should get some testing.

Re: flattening the layouts.  A good question for @mshinwell.  My instinct is that the job of flattening products is part of "unarization", and that task belongs to the `lambda` -> `flambda` pass, so better to keep the product structure here.